wgengine/magicsock/magicsock.go

```diff
@@ -276,6 +276,16 @@ type Conn struct {
 	// discoKey is the current disco private and public keypair for this conn.
 	discoKey *key.DiscoKey
 
+	// discoKeyCreatedAt is when the current disco key was created.
+	// Used for both rate limiting rotations (ensuring keys are old enough to rotate)
+	// and for tracking when to cleanup the old key after the grace period.
+	discoKeyCreatedAt atomic.Pointer[time.Time]
+
+	// oldDiscoKey is the previous disco private key, kept during the grace
+	// period after a rotation to allow peers to decrypt messages sent with
+	// the old key until they receive the new key from the control plane.
+	oldDiscoKey atomic.Pointer[key.DiscoPrivate]
```

Comment:
Concern (blocking): If the disco key is rotated twice in quick succession, do we want the original disco key to still be valid for the grace period? My expectation is yes?